### PR TITLE
feat(drive)!: provide protocol version in epoch info query

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

We would like to persist the protocol version for each epoch to have historical information of version upgrades in drive state.
 It would make sense to provide this information in epoch info queries as well.

## What was done?

- Persisted next epoch protocol version to the upcoming epoch
- Added protocol version of epoch to the get epochs info response
- Remove current and next protocol version from misc tree  

## How Has This Been Tested?

Querying epoch info for a particular epoch and asserting expected protocol version in the protocol upgrade strategy tests
 

## Breaking Changes

Addition key-value in Epoch Tree that makes previously created state is not valid.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 1]
```diff
@@ -8,31 +8,31 @@ import "google/protobuf/timestamp.proto";
...
   rpc getDocuments(GetDocumentsRequest) returns (GetDocumentsResponse);
   rpc getIdentitiesByPublicKeyHashes(GetIdentitiesByPublicKeyHashesRequest)
-      returns (GetIdentitiesByPublicKeyHashesResponse);
+    returns (GetIdentitiesByPublicKeyHashesResponse);
   rpc getIdentityByPublicKeyHash(GetIdentityByPublicKeyHashRequest)
-      returns (GetIdentityByPublicKeyHashResponse);
+    returns (GetIdentityByPublicKeyHashResponse);
   rpc waitForStateTransitionResult(WaitForStateTransitionResultRequest)
-      returns (WaitForStateTransitionResultResponse);
+    returns (WaitForStateTransitionResultResponse);
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:33]: Why is the formatting changing?

File: packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs [hunk 2]
```diff
@@ -75,7 +77,8 @@ impl Drive {
         query.set_subquery(subquery);
         let path_query = PathQuery::new(
             pools_vec_path(),
-            SizedQuery::new(query, Some(count * 4), None),
+            // The multipler must be equal to requested keys count
```
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs:80]: multiplier

File: packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs [hunk 3]
```diff
@@ -212,13 +215,35 @@ impl Drive {
...
+                else {
+                    return Some(Err(Error::Drive(DriveError::UnexpectedElementType(
+                        "start time must be an item",
+                    ))));
+                };
+
+                let protocol_version_bytes: [u8; 4] =
+                    match encoded_protocol_version.as_slice().try_into().map_err(|_| {
+                        Error::Drive(DriveError::CorruptedSerialization(
+                            "core block height must be 4 bytes for a u32".to_string(),
```
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs:230]: Wrong error message

File: packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs [hunk 4]
```diff
@@ -212,13 +215,35 @@ impl Drive {
 
                 let fee_multiplier = f64::from_be_bytes(fee_multiplier_bytes);
 
+                let protocol_version_element = inner_map.get(&KEY_PROTOCOL_VERSION.to_vec())?;
+
+                let Some(Element::Item(encoded_protocol_version, _)) = protocol_version_element
+                else {
+                    return Some(Err(Error::Drive(DriveError::UnexpectedElementType(
+                        "start time must be an item",
```
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/system/verify_epoch_infos/v0/mod.rs:223]: Wrong error message.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
We should add a test to make sure it works. Maybe in sdk.

**Comment by shumkov** [comment]
I've added query asserts to existing version upgrade tests

**Review by QuantumExplorer — APPROVED** [review]

**Review by QuantumExplorer — DISMISSED** [review]

**Review by QuantumExplorer — APPROVED** [review]

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**